cow_instrument/PointPathComponent.h

```diff
@@ -0,0 +1,111 @@
+#ifndef POINT_PATH_COMPONENT_H
+#define POINT_PATH_COMPONENT_H
+
+#include "PathComponent.h"
+#include "V3D.h"
+#include "Component.h"
+#include "IdType.h"
+
+/**
+ * CRTP type to do boiler plate work of creating PathComponents that behave like
+ * points from a neutronic perspective.
+ */
+template <typename T> class PointPathComponent : public PathComponent {
+
+public:
+  PointPathComponent(V3D pos, ComponentIdType id);
+  virtual V3D getPos() const override;
+  virtual void deltaPos(const V3D &pos) override;
+
+  virtual bool equals(const Component &other) const override;
+  virtual void registerContents(
+      std::vector<const Detector *> &detectorLookup,
+      std::vector<const PathComponent *> &pathLookup) const override;
+  virtual ComponentIdType componentId() const override;
+  virtual std::string name() const override;
+
+  double length() const override;
+  V3D entryPoint() const override;
+  V3D exitPoint() const override;
+  virtual ~PointPathComponent() {}
+  virtual PointPathComponent<T> *clone() const override;
+
+  bool operator==(const PointPathComponent<T> &other) const;
+  bool operator!=(const PointPathComponent<T> &other) const;
+
+private:
+  V3D m_pos;
+  ComponentIdType m_componentId;
+};
+
+template <typename T>
+PointPathComponent<T>::PointPathComponent(V3D pos, ComponentIdType id)
+    : m_pos(pos), m_componentId(id) {}
+
+template <typename T> V3D PointPathComponent<T>::getPos() const {
+  return m_pos;
+}
+
+template <typename T> void PointPathComponent<T>::deltaPos(const V3D &pos) {
+  m_pos[0] += pos[0];
+  m_pos[1] += pos[1];
+  m_pos[2] += pos[2];
+}
+
+template <typename T>
+PointPathComponent<T> *PointPathComponent<T>::clone() const {
+  return new PointPathComponent<T>(*this);
```

Comment:
I did have that originally. The problem is that at the time that the compiler processes the CRTP base `PointPathComponent<T>` it does not know that `T` is also derived from `Component`. I think a better fix for this would be for me to push the clone definitions down to the final types, `T` in this case and not have the generic implementation.